### PR TITLE
Handle null proxy hosts

### PR DESCRIPTION
**Pull request by shadab-geo**
Issue: https://github.com/geosolutions-it/http-proxy/issues/65

### CODE REVIEW

**Review by simboss — CHANGES_REQUESTED** [review]
Where is the unit test for this change?

No unit test no party.

**Comment by shadab-geo** [comment]
> Where is the unit test for this change?
> 
> No unit test no party.

agree😄 👍 
Added JUnit test

File: src/main/java/it/geosolutions/httpproxy/HTTPProxy.java [hunk 1]
```diff
@@ -209,12 +209,18 @@ public HttpRoute determineRoute(
...
+                            return new HttpRoute(target);
+                        else
+                            return new HttpRoute(target, null, httpHost,
+                                    false);
                     } else {
                         LOGGER.debug("Setting https scheme");
-                        return new HttpRoute(target, null, httpsHost,
-                                true);
+                        if (httpsHost == null)
+                            return new HttpRoute(target);
```
Comment (taba90) [inline src/main/java/it/geosolutions/httpproxy/HTTPProxy.java:220]: The HttpRoute constructor accepting a single HttpHost is documented from the java docs to be creating a new direct insecure route. I guess then that when the protocol is https then we should use this other constructor with the secure flag to true:
public HttpRoute([HttpHost](https://hc.apache.org/httpcomponents-core-ga/httpcore/apidocs/org/apache/http/HttpHost.html?is-external=true) target,
[InetAddress](https://download.oracle.com/javase/1.5.0/docs/api/java/net/InetAddress.html?is-external=true) local,
boolean secure);

File: src/main/java/it/geosolutions/httpproxy/HTTPProxy.java [hunk 2]
```diff
@@ -209,12 +209,18 @@ public HttpRoute determineRoute(
                     LOGGER.info("Returning proxy route");
                     if (target.getSchemeName().equals("http")) {
                         LOGGER.debug("Setting http scheme");
-                        return new HttpRoute(target, null, httpHost,
```
Comment (taba90) [inline src/main/java/it/geosolutions/httpproxy/HTTPProxy.java:212]: The code nesting here is a lot. I would refact this code a bit to provide a separate method doing the creation of the right HttpRoute object or I would turn the anonymous class into a nested one.

**Review by taba90 — CHANGES_REQUESTED** [review]
Overall looks good. Few modifications required, see the comments.

File: src/main/java/it/geosolutions/httpproxy/HTTPProxy.java (hunk 1 above)
Comment (shadab-geo) [reply src/main/java/it/geosolutions/httpproxy/HTTPProxy.java]: updated the code

File: src/main/java/it/geosolutions/httpproxy/HTTPProxy.java (hunk 2 above)
Comment (shadab-geo) [reply src/main/java/it/geosolutions/httpproxy/HTTPProxy.java:212]: created a new method for proxy HTTPRoute object creation

**Comment by shadab-geo** [comment]
> Overall looks good. Few modifications required, see the comments.

Thanks, @taba90 
have applied the feedback

**Comment by taba90** [comment]
thanks! merging

**Pull request merged**